linkedin/datahub
# feat(tag): adding search for tags in gms layer

Test plan:
- run local gms & mae-consumer
- run `./docker/dev.sh`
- run `datahub ingest -c ./examples/recipes/example_to_datahub_rest.yml`

verify:
```
(venv) ➜  curl 'http://localhost:8080/tags?q=search&input=legacy'

{"metadata":{"urns":["urn:li:tag:Legacy"],"searchResultMetadatas":[]},"elements":[{"ownership":{"owners":[{"owner":"urn:li:corpuser:jdoe","type":"DATAOWNER"}],"lastModified":{"actor":"urn:li:corpuser:jdoe","time":1581407189000}},"name":"Legacy","description":"Indicates the dataset is no longer supported"}],"paging":{"count":10,"start":0,"total":1,"links":[]}}%  
```

verify:
```
(venv) ➜ curl "http://localhost:8080/tags?action=autocomplete" -d '{"query": "lega", "field": "name", "limit": 10, "filter": {"criteria": []}}' -X POST -H 'X-RestLi-Protocol-Version: 2.0.0' | jq

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   126  100    51  100    75   3187   4687 --:--:-- --:--:-- --:--:--  7875
{
  "value": {
    "suggestions": [
      "Legacy"
    ],
    "query": "lega"
  }
}
```

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Docs related to the changes have been added/updated (if applicable)
